library/Faktory/Ent/Batch.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE DerivingVia #-}
+
+-- | Support for the @BATCH@ command (Enterprise only)
+--
+-- <https://github.com/contribsys/faktory/wiki/Ent-Batches>
+--
+-- Batches allow multiple Jobs to be enqueued as a group, with a description
+-- (visible in the admin UI) and Jobs attached to run on completion of all Jobs
+-- within the group (always, or only if all were successful).
+--
+-- Usage:
+--
+-- @
+-- -- Build a Job to run at completion of the Batch. Arguments are the same as
+-- -- you would pass to 'perform' the Job.
+-- onComplete <- buildJob mempty producer myJob
+--
+-- 'runBatchT' ('complete' onComplete <> 'description' "My Batch") producer $ do
+--   -- Use 'batchPerform' instead of 'perform'
+--   void $ 'batchPerform' mempty producer myBatchedJob1
+--   void $ 'batchPerform' mempty producer myBatchedJob2
+-- @
+--
+-- __/NOTE__: This module does not support batched Jobs dynamically adding more
+-- Jobs to the Batch. PRs welcome.
+--
+module Faktory.Ent.Batch
+    (
+    -- * Options
+      BatchOptions
+    , description
+    , complete
+    , success
+
+    -- * Running
+    , runBatchT
+    , BatchT
+    , batchPerform
+    )
+where
+
+import Faktory.Prelude
+
+import Control.Monad.Reader
+import Data.Aeson
+import Data.Aeson.Casing
+import Data.ByteString.Lazy as BSL
+import Data.Semigroup (Last(..))
+import Data.Semigroup.Generic
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+import Faktory.Client
+import Faktory.Job
+import Faktory.Producer
+import GHC.Generics
+import GHC.Stack
+
+newtype BatchT a = BatchT (ReaderT BatchId IO a)
```

Comment:
`BatchT` makes this sounds like a transformer, should this be:

```haskell
newtype BatchT m a = BatchT (ReaderT BatchId m a)
```
?

It looks like most everything else in this library is in concrete `IO` though

Reply:
It is a transformer, over `IO`, right? I didn't know if `T` still applied. Usually no `T` implies `Identity`.

Reply:
To clarify, I'm riding the `IO` wave of the library at the moment -- though going `MonadIO m` _generally_ is the back of my mind. So, given that, it's a naming question: `runBatch`?

Reply:
Yeah, I would either have `runBatch :: Batch a -> ... -> IO a` or `runBatchT :: ... => BatchT m a -> ... m a`

Reply:
Coolio, will rename.